rust/src/encodings/plain.rs

```diff
@@ -70,12 +70,24 @@ impl<'a, R: Read + Seek, T: NativeType> Decoder<T> for PlainDecoder<'a, R> {
                         Err(_) => unreachable!(),
                     };
                     *slot = T::from_le_bytes(a);
-                    Ok(())
+                    Ok::<(), Error>(())
                 })?;
             let arr = PrimitiveArray::from_vec(buffer);
             Ok(Box::new(arr))
         }
     }
+}
+
+impl<'a, R: Read + Seek, T: NativeType> Decoder for PlainDecoder<'a, R, T> {
+    // type ArrowType = T;
+
+    fn decode(&mut self, offset: i32, length: &Option<i32>) -> Result<Box<dyn Array>> {
+        self._decode(offset, length).map(|x| {
+            // TODO no idea why we need this to compile
+            let new_box: Box<dyn Array> = Box::new(*x);
```

Comment:
Because `_decode` returns `PrimitiveArray<T>` not `<Array>`?

Reply:
I deliberately keep the concrete type parameters of `_decode` for use in both `decode` and `value`, but I think the way to convert types is not decent enough, better recommendations?